boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -59,8 +59,8 @@ ifeq ($(CONFIG_CXX_EXCEPTION),)
   ARCHCPUFLAGSXX += -fno-exceptions
 endif
 ARCHPICFLAGS = -fpic
-ARCHWARNINGS = -Wall -Wstrict-prototypes -Wshadow -Wundef
-ARCHWARNINGSXX = -Wall -Wshadow -Wundef
+ARCHWARNINGS = -Wstrict-prototypes -Wundef
+ARCHWARNINGSXX = -Wundef
```

Comment:
the removal of -Wshadow is explained in the commit message.
but the removal of -Wall is not. can you explain a bit?

Reply:
Yes, it isn't required for libc++, but I have saw other 3rd party library can't be the warning free with -Wall, and it's hard to disable -Wall by CFLAGS. Do you hit some special issue without -Wall?

Reply:
> Yes, it isn't required for libc++, but I have saw other 3rd party library can't be the warning free with -Wall, and it's hard to disable -Wall by CFLAGS.

how is it hard to disable?
-Wall is basically a set of -Wxxx. you can disable each of them by -Wno-xxx.

> Do you hit some special issue without -Wall?

no special issue. i just prefer to enable it as it's often useful.

Reply:
I mean there isn't -Wno-all to restore to the default.